fix(logging): have default log level be 'error'

JohnVillalovos opened the pull request:
If no logging level is set then default to 'error'

Previously it was 'none'. In that case critical errors could be happening but none of them would show up in the log.

Change logging levels to lowercase to fix bug as uppercase is incorrect.

Enable default values to work for config items.

lucs7 commented on lib/Config/ConfigKeys.php:
```diff
@@ -414,7 +414,7 @@ class ConfigKeys
         'key' => 'logging.level',
         'type' => 'string',
-        'default' => 'none',
+        'default' => 'INFO',
```
This would also requirea nother edit to Log.php as the case currently only covers 'debug' and 'error'
```php
        if ($log_level != 'none') {
            $log_folder = Configuration::Instance()->GetKey(ConfigKeys::LOGGING_FOLDER);
            $log_sql = Configuration::Instance()->GetKey(ConfigKeys::LOGGING_SQL, new BooleanConverter());
            switch ($log_level) {
                case 'debug':
                    $this->logger->pushHandler(new StreamHandler($log_folder.'/app.log', Logger::DEBUG));
                    break;
                case 'error':
                    $this->logger->pushHandler(new StreamHandler($log_folder.'/app.log', Logger::ERROR));
                    break;
            }
            $this->logger->pushProcessor(new WebProcessor());
        }
        if ($log_sql) {
            $this->sqlLogger->pushHandler(new StreamHandler($log_folder.'/sql.log', Logger::ERROR));
        }
```

lucs7 committed:
fix: use default logging level of 'error'

If no logging level is set then default to 'error'

Previously it was 'none'. In that case critical errors could be
happening but none of them would show up in the log.

We currently only support logging levels of 'none', 'debug', and
'error'. Remove unsupported levels.

Change logging levels to lowercase to fix bug as uppercase is incorrect.

lucs7 committed:
fix: add default value handling in ConfigurationFile::GetKey method
